src/NServiceBus.Core/Routing/FileBasedDynamicRouting/FileRoutingTableParser.cs

```diff
@@ -0,0 +1,38 @@
+namespace NServiceBus.Routing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Xml.Linq;
+
+    class FileRoutingTableParser
+    {
+        public IEnumerable<EndpointInstance> Parse(XDocument document)
+        {
+            var root = document.Root;
+            var endpointElements = root.Descendants("endpoint");
+
+            var instances = new List<EndpointInstance>();
+
+            foreach (var e in endpointElements)
+            {
+                var nameAttribute = e.Attribute("name");
+                if (nameAttribute == null)
+                {
+                    throw new Exception("Endpoint does not have a name.");
+                }
+                var endpointName = new Endpoint(nameAttribute.Value);
+                foreach (var i in e.Descendants("instance"))
+                {
+                    var discriminatorAttribute = i.Attribute("discriminator");
```

Comment:
Should we changed the attribute name to scaleOutDiscriminator to express it's current usage?